wms_connector/models/stock_warehouse.py

```diff
@@ -0,0 +1,113 @@
+    wms_product_filter_id = fields.Many2one(
+        "ir.filters",
+        default=lambda r: r.env.ref("wms_connector.default_empty_filter"),
+    )
+    wms_picking_ar_filter_id = fields.Many2one(
+        "ir.filters",
+        default=lambda r: r.env.ref("wms_connector.default_empty_filter"),
+    )
+    wms_picking_prp_filter_id = fields.Many2one(
+        "ir.filters",
+        default=lambda r: r.env.ref("wms_connector.default_empty_filter"),
+    )
+    wms_product_sync_ids = fields.One2many("product.product", "warehouse_id")
+
+    def _inverse_active_wms_sync(self):
+        for rec in self:
+            if rec.active_wms_sync:
+                rec._activate_crons_tasks()
+            else:
+                rec._deactivate_crons_tasks()
+
+    def _activate_crons_tasks(self):
+        for kind in TASKS:
+            task_field_name = TASK_FIELDNAME.format(kind[0])
+            task = getattr(self, task_field_name)
+            if task:
+                task.active = True
+            else:
+                setattr(
+                    self,
+                    task_field_name,
+                    self.env["attachment.synchronize.task"].create(
+                        {
+                            "name": "WMS task for {} ({})".format(self.name, kind[1]),
+                            "method_type": "export",
+                            "filepath": "OUT/",
+                            "backend_id": self.env.ref(
+                                "storage_backend.default_storage_backend"
+                            ),
+                        }
+                    ),
+                )
+            cron_field_name = CRON_FIELDNAME.format(kind[0])
+            cron = getattr(self, cron_field_name)
+            if cron:
+                cron.active = True
+            else:
+                setattr(
+                    self,
+                    cron_field_name,
+                    self.env["ir.cron"].create(
+                        {
+                            "name": "WMS cron for {} ({})".format(self.name, kind[1]),
+                            "active": False,
+                            "interval_type": "days",
+                            "interval_number": 1,
+                            "model_id": self.env.ref(
+                                "attachment_synchronize.model_attachment_synchronize_task"
+                            ).id,
+                            "state": "code",
+                            "code": "",  # TODO
+                        }
+                    ),
+                )
+
+    def _deactivate_crons_tasks(self):
+        for kind in TASKS:
+            setattr(self, TASK_FIELDNAME.format(kind[0]), False)
+            setattr(self, CRON_FIELDNAME.format(kind[0]), False)
+
+    def action_open_flows(self):
+        raise NotImplementedError
+        return {"type": "ir.action"}
+
+    def refresh_wms_products(self):
+        for rec in self:
+            rec.wms_product_sync_ids.unlink()
```

Comment:
do not drop everything, it will be costly and you lose the sync information
Just search all product then
- create the missing binding
- unlink the binding that do not match the list of product